core/backend/src/IModelDb.ts

```diff
@@ -853,6 +853,22 @@ export abstract class IModelDb extends IModel {
     return this[_nativeDb].restartTxnSession();
   }
 
+  /** Drops schemas from an array of schema
+* @param schemaNames Array of schema names to drop
+* @alpha
+*/
+  public async dropSchemas(schemaNames: string[]): Promise<void> {
+    if (this[_nativeDb].getITwinId() !== Guid.empty) {
+      await this.acquireSchemaLock();
+    }
+    try {
+      await this[_nativeDb].dropSchemas(schemaNames);
+    } catch (error) {
+      Logger.logError(loggerCategory, `Failed to drop schemas: ${error}`);
```

Comment:
call abandonChanges() here do not leave connection in a inconsistent state. What if user call saveChanges() after this point.